service/history/task/cross_cluster_target_task_executor.go

```diff
@@ -239,37 +239,83 @@ func (t *crossClusterTargetTaskExecutor) executeApplyParentClosePolicyTask(
 		return nil, errMissingTaskRequestAttributes
 	}
 
-	for _, childAttrs := range attributes.ApplyParentClosePolicyAttributes {
+	response := types.CrossClusterApplyParentClosePolicyResponseAttributes{}
+	var anyErr error
+
+	for _, child := range attributes.Children {
+		// DON'T RETURN ERROR INSIDE THIS LOOP, CONVERT THEM AND ASSIGN IT TO EACH CHILD'S FailedCause
+		childAttrs := child.Child
+		if child.Status != nil && child.Status.Completed {
+			// This means we are in retry and this child was already successfully processed before
+			response.ChildrenStatus = append(
+				response.ChildrenStatus,
+				&types.ApplyParentClosePolicyResult{
+					Child:       childAttrs,
+					FailedCause: nil,
+				})
+			continue
+		}
+		if child.Status != nil && child.Status.FailedCause != nil && !child.Status.Retriable {
+			// This means child failed with a non-retriable error before, there's nothing to do
+			response.ChildrenStatus = append(
+				response.ChildrenStatus,
+				&types.ApplyParentClosePolicyResult{
+					Child:       childAttrs,
+					FailedCause: child.Status.FailedCause,
+				})
+			continue
+		}
+		if child.Status == nil {
+			child.Status = &types.ApplyParentClosePolicyStatus{}
+		}
+		child.Status.Completed = false
+		scope := t.metricsClient.Scope(metrics.CrossClusterSourceTaskApplyParentClosePolicyScope)
+		var failedCause *types.CrossClusterTaskFailedCause
+		retriable := false
+
 		targetDomainName, err := t.verifyDomainActive(childAttrs.ChildDomainID)
-		if err != nil {
-			return nil, err
+		if err == nil {
+			err = applyParentClosePolicy(
+				ctx,
+				t.historyClient,
+				childAttrs.ChildDomainID,
+				targetDomainName,
+				childAttrs.ChildWorkflowID,
+				childAttrs.ChildRunID,
+				*childAttrs.ParentClosePolicy,
+			)
 		}
 
-		scope := t.metricsClient.Scope(metrics.CrossClusterSourceTaskApplyParentClosePolicyScope)
-		err = applyParentClosePolicy(
-			ctx,
-			t.historyClient,
-			childAttrs.ChildDomainID,
-			targetDomainName,
-			childAttrs.ChildWorkflowID,
-			childAttrs.ChildRunID,
-			*childAttrs.ParentClosePolicy,
-		)
 		switch err.(type) {
-		case *types.EntityNotExistsError, *types.WorkflowExecutionAlreadyCompletedError, *types.CancellationAlreadyRequestedError:
+		case *types.EntityNotExistsError,
+			*types.WorkflowExecutionAlreadyCompletedError,
+			*types.CancellationAlreadyRequestedError:
 			// expected error, no-op
-			break
+			scope.IncCounter(metrics.ParentClosePolicyProcessorSuccess)
```

Comment:
I am thinking maybe we should:
- emit those metrics at the source cluster
- let the source cluster determine how to handle those errors. The target cluster just reports the result/failed cause as is.
What's your thought? 
Those are not blocking, we can do them later if we want to.